catlearn/relation_cache.py

```diff
@@ -492,3 +494,35 @@ def match(
                 result_graph[src][tar].pop(NegativeMatch(best_match), None)
 
         return result_graph
+
+    def sort_relations(
+            self,
+            src: Optional[NodeType] = None, tar: Optional[NodeType] = None,
+            labels: Optional[FrozenSet[ArrowType]] = None
+    ) -> OrderedDict[Tuple[NodeType, NodeType, FrozenSet[ArrowType]], float]:
+        """
+        Given a result graph from a match, sort available relations matching
+        the specific properties (in terms of score):
+        - if src is given, will only look at relations with this source
+        - if tar is given, will only look at relations with this target
+        - if labels is given, will only look at the score corresponding to
+        the sum for these labels. Otherwise will list all possible
+        individual label scores.
+        """
+        arrows = self.arrows(src, tar, include_non_causal=False)
+
+        if labels is None:
+            options = {
+                (arr[0], arr[-1], frozenset((label,))): float(kl_match(
+                    self[arr], self.label_universe[label]))
+                for (arr, label) in product(arrows, self.label_universe)}
+        else:
+            options = {
+                (arr[0], arr[-1], frozenset(labels)): float(sum(
+                    kl_match(self[arr], self.label_universe[label])
+                    for label in labels))
+                for arr in arrows}
+
+        return OrderedDict[
```

Comment:
I didn't know you could use a typing annotations to directly instanciate an object. Neat!